Toolkit/ArcGISToolkit/TemplatePickerViewController.swift

```diff
@@ -179,9 +179,8 @@ public class TemplatePickerViewController: TableViewController {
                     // update info with swatch
                     infos[index].swatch = image
                     
-                    // reload index where that info currently is
-                    let indexPath = self.indexPath(for: info)
-                    self.tableView.reloadRows(at: [indexPath], with: .automatic)
+                    // The size of the table view is still not finalized at this point, so we can't use `reloadRows(...)`.
+                    self.tableView.reloadData()
```

Comment:
`reloadRows` provides a better UX when just updating the swatch. Is it possible to conditionalize this fix? I don't remember the specifics of this particular function, but if it reloads the entire table view every time a swatch comes in, that seems to be inefficient.

Is the indexPath nil or invalid in the crash case? If so, can you just bail?

Reply:
Hmm when I was testing I don't think `indexPath` was ever nil, I think it was referring to an `indexPath` in the table that did not exist at the time. Let me double check that

Reply:
The index path in question is calculated based on the feature table and feature template info and their locations in the respective data sources, so there should always be a valid indexPath.

The `loadInfosAndCreateSwatches` where the code in question resides is only called once from `viewDidLoad`, so the overhead of using `reloadData()` won't be too bad.

Reply:
So the method returning the `indexPath` returns a non-optional. But I checked and the `indexPath` is referring to a section one that is not available when it runs `reloadRows`

Reply:
If you try to find the number of rows in the section (using `tableView.numberOfRows`) that is assigned to `indexPath` it crashes because that section is out of bounds of the current table view.

Reply:
Alright let me check that out

Reply:
After debugging this with Zach and looking at the code a bit more, I do think just checking to make sure that the index path exists is a valid fix for this issue.